### PR TITLE
Introducing FileSystemHelper interface

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
This interface handles filesystem functionality used mainly by worker In the latter PRs it will be updated to extend the support of various in-tree removal scenarious. In addition it will allow for better testing of the functionality related to deleting/verifying/copying files

### CODE REVIEW

File: cmd/worker/funcs_kmod.go [hunk 1]
```diff
@@ -17,7 +18,8 @@ func rootFuncPreRunE(cmd *cobra.Command, args []string) error {
 	logger.Info("Starting worker", "version", Version, "git commit", commit)
 
 	mr := worker.NewModprobeRunner(logger)
-	w = worker.NewWorker(mr, logger)
+	fh := utils.NewFSHelper(logger)
```
Comment (ybettan) [inline cmd/worker/funcs_kmod.go:21]:
```suggestion
	fsh := utils.NewFSHelper(logger)
```

File: internal/utils/fshelper.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			fh.logger.Info("Removing dst file", "file", fileToRemove)
+			err = os.Remove(fileToRemove)
+			if err != nil {
+				fh.logger.Info(WarnString("failed to delete file"), "file", fileToRemove, "error", err)
+			}
+		}
+		return nil
+	})
+	return err
+}
```
Comment (ybettan) [inline internal/utils/fshelper.go:47]:
```suggestion
func (fh *fsHelper) RemoveSrcFilesFromDst(srcDir, dstDir string) error {
	err := filepath.Walk(srcDir, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		if !info.IsDir() {
			relPath, err := filepath.Rel(srcDir, path)
			if err != nil {
				fh.logger.Info(WarnString("failed to get relative path"), "srcDir", srcDir, "path", path, "error", err)
				return nil
			}
			fileToRemove := filepath.Join(dstDir, relPath)
			fh.logger.Info("Removing dst file", "file", fileToRemove)
			err = os.Remove(fileToRemove)
			if err != nil {
				fh.logger.Info(WarnString("failed to delete file"), "file", fileToRemove, "error", err)
			}
		}
		return nil
	})
	if err != nil {
	    return fmt.Errorf("failed to remove file %s/* from %s\n", srcDir, dstDir)
	}
	return nil
}
```

File: internal/utils/fshelper.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			fileToRemove := filepath.Join(dstDir, relPath)
+			fh.logger.Info("Removing dst file", "file", fileToRemove)
+			err = os.Remove(fileToRemove)
+			if err != nil {
+				fh.logger.Info(WarnString("failed to delete file"), "file", fileToRemove, "error", err)
+			}
+		}
+		return nil
+	})
+	return err
```
Comment (ybettan) [inline internal/utils/fshelper.go:46]:
```suggestion
	if err != nil {
	    return fmt.Errorf("failed to remove file %s/* from %s\n", srcDir, dstDir)
	}
	return nil
```

File: internal/utils/fshelper.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package utils
```
Comment (ybettan) [inline internal/utils/fshelper.go:1]: Can we have a full name for this file? `filesystemHelper.go`?

File: cmd/worker/funcs_kmod.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply cmd/worker/funcs_kmod.go:21]: Done

File: internal/utils/fshelper.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/utils/fshelper.go:46]: done

File: internal/utils/fshelper.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply internal/utils/fshelper.go:1]: changed the name. Regarding its own package: we currently have a lot of minor interfaces/functionalities in utils. We can discuss if we want to dismantle this package in a later PRs

Comment (ybettan) [reply internal/utils/fshelper.go]: OK then.

**Pull request merged**